src/agni/server/communication/MessageSender.java

```diff
@@ -1,15 +1,24 @@
 package agni.server.communication;
 
-import java.net.Socket;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.SocketChannel;
 
 
 public class MessageSender implements I_MessageSender {
-    
+    private ChannelList channelList;
+  
     public MessageSender(ChannelList channels) {
-        // TODO
+      channelList = channels;
     }
 
     public void sendMessage(String ipAddress, byte[] message) {
-        // TODO
+       SocketChannel currentChannel = channelList.getChannel(ipAddress);
+       ByteBuffer outBuffer = ByteBuffer.wrap(message);
+       try {
+    currentChannel.write(outBuffer);
+       } catch (IOException e) {
+    e.printStackTrace();
+       }
```

Comment:
Instead of just printing the stack trace, let's print a more meaningful message. I have a feeling we'll be getting into this `catch` alot.